[LabelGroup] new Tag feature

CApopsicle opened the pull request:
`<Label.Group />` component new tag feature
http://jira.cepave.com/browse/OWL-1434

![kapture 2017-03-06 at 16 33 36](https://cloud.githubusercontent.com/assets/5334755/23646965/5cfcade6-034e-11e7-9f79-d43cd716bd0c.gif)

- [x] component
- [x] test specs
- [x] doc

cc @rwu823

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cepave-f2e/vue-owl-ui/139)
<!-- Reviewable:end -->

CApopsicle committed:
[LabelGroup] add new tag feature

CApopsicle committed:
[LabelGroup] add new tag feature: test spec

CApopsicle committed:
[LabelGroup] add new tag feature: doc

desmond9319 commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented:
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/components/label/index.js, line 159 at r1](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5sZD1_40JR7SlxF_g:-Kf5sZD1_40JR7SlxF_h:bxg2i2z) ([raw file](https://github.com/cepave-f2e/vue-owl-ui/blob/037f3915665ee096b0b02efaa09147e0b65de7ea/src/components/label/index.js#L159)):*
> ```JavaScript
>         [s.newTag]: true,
>         [s.showInput]: showInput
>       }
> ```

永遠都是 `true` 可放進 array 裡就好 

```jsx
const styles = [s.label, s.newTage, {
  [s.showInput]: showInput
}]
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[src/components/label/index.js, line 205 at r1](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5tCFhmdu0j42a2EJm:-Kf5tCFhmdu0j42a2EJn:b-9q2p9u) ([raw file](https://github.com/cepave-f2e/vue-owl-ui/blob/037f3915665ee096b0b02efaa09147e0b65de7ea/src/components/label/index.js#L205)):*
> ```JavaScript
>         if (preventDuplicate) {
>           let duplicate = false
>           this.labelData.forEach((data, idx) => {
> ```

找到即停，用 `.find`  效率較好

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5u2EqTBkdnExKfwjY:-Kf5u2EqTBkdnExKfwjZ:b-blpbxe) (no related file):*
```js
this.$emit('change', this.labelData)
```

這如果出現在很多地方，最近發現一招放進 watch 裡就好， 一勞永逸

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented:
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented on src/components/label/index.js:
```diff
@@ -126,35 +143,109 @@ Label.Group = {
+
+        if (preventDuplicate) {
+          let duplicate = false
+          this.labelData.forEach((data, idx) => {
```
`.find`

rwu823 replied:
!fix

rwu823 replied:
!fixed

rwu823 replied:
!fix

chuanxd commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5u2EqTBkdnExKfwjY:-KfAFnB09sQ3SgvVR1tv:bmeyrwp) (no related file):*
<details><summary><i>Previously, rwu823 (Rocky Wu) wrote…</i></summary><blockquote>

```js
this.$emit('change', this.labelData)
```

這如果出現在很多地方，最近發現一招放進 watch 裡就好， 一勞永逸
</blockquote></details>

:+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

CApopsicle commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[src/components/label/index.js, line 205 at r1](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-KeblRgfZF3BopPQ7hEg-r1-205:-KfAJSW77HFNZRnG7Zay:b-2cquku) ([raw file](https://github.com/cepave-f2e/vue-owl-ui/blob/037f3915665ee096b0b02efaa09147e0b65de7ea/src/components/label/index.js#L205)):*
<details><summary><i>Previously, rwu823 (Rocky Wu) wrote…</i></summary><blockquote>

!fix
</blockquote></details>

這是什麼啊XD 有github的icon

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

CApopsicle commented:
Review status: 4 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[a discussion](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5u2EqTBkdnExKfwjY:-KfAz-ysAnBlPLu5MXvF:b-7ql1td) (no related file):*
<details><summary><i>Previously, chuanxd (Timmy Chang) wrote…</i></summary><blockquote>

:+1: 
</blockquote></details>

fixed

---

*[src/components/label/index.js, line 159 at r1](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5sZD1_40JR7SlxF_g:-KfAz15H2eYRBVNOv3ix:b-7ql1td) ([raw file](https://github.com/cepave-f2e/vue-owl-ui/blob/037f3915665ee096b0b02efaa09147e0b65de7ea/src/components/label/index.js#L159)):*
<details><summary><i>Previously, rwu823 (Rocky Wu) wrote…</i></summary><blockquote>

永遠都是 `true` 可放進 array 裡就好 

```jsx
const styles = [s.label, s.newTage, {
  [s.showInput]: showInput
}]
```
</blockquote></details>

fixed

---

*[src/components/label/index.js, line 205 at r1](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139#-Kf5tCFhmdu0j42a2EJm:-KfAz2Foh_nCPxnzPMGP:b-7ql1td) ([raw file](https://github.com/cepave-f2e/vue-owl-ui/blob/037f3915665ee096b0b02efaa09147e0b65de7ea/src/components/label/index.js#L205)):*
<details><summary><i>Previously, rwu823 (Rocky Wu) wrote…</i></summary><blockquote>

找到即停，用 `.find`  效率較好
</blockquote></details>

fixed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->

rwu823 commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cepave-f2e/vue-owl-ui/139)*
<!-- Sent from Reviewable.io -->
